src/Support/SupportData.php

```diff
@@ -0,0 +1,930 @@
+
+			if ( ! isset( $stylesheets[ $i ]['group'] ) || 'amp-custom' !== $stylesheets[ $i ]['group'] || ! empty( $stylesheets[ $i ]['duplicate'] ) ) {
+				continue;
+			}
+
+			$max_final_size = max( $max_final_size, $stylesheets[ $i ]['final_size'] );
+			if ( $stylesheets[ $i ]['included'] ) {
+				$included_final_size    += $stylesheets[ $i ]['final_size'];
+				$included_original_size += $stylesheets[ $i ]['original_size'];
+
+				if ( $included_final_size >= $style_custom_cdata_spec['max_bytes'] ) {
+					$stylesheets[ $i ]['status'] = $excessive_status;
+				} else {
+					$stylesheets[ $i ]['status'] = $included_status;
+				}
+			} else {
+				$excluded_final_size    += $stylesheets[ $i ]['final_size'];
+				$excluded_original_size += $stylesheets[ $i ]['original_size'];
+				$excluded_stylesheets ++;
+				$stylesheets[ $i ]['status'] = $excluded_status;
+			}
+		}
+
+		$percentage_budget_used = ( ( $included_final_size + $excluded_final_size ) / $style_custom_cdata_spec['max_bytes'] ) * 100;
+		$response               = [
+			'css_size_before'       => intval( $included_original_size + $excluded_original_size ),
+			'css_size_after'        => intval( $included_final_size + $excluded_final_size ),
+			'css_size_excluded'     => intval( $excluded_stylesheets ),
+			'css_budget_percentage' => round( $percentage_budget_used, 1 ),
+		];
+
+		return $response;
+
+	}
+
+	/**
+	 * To get home url of the site.
+	 * Note: It will give home url without protocol.
+	 *
+	 * @since 2.2
+	 *
+	 * @return string Home URL.
+	 */
+	public static function get_home_url() {
+
+		$home_url = home_url();
+		$home_url = strtolower( trim( $home_url ) );
+
+		$http_protocol = wp_parse_url( $home_url, PHP_URL_SCHEME );
+
+		$home_url = str_replace( "$http_protocol://", '', $home_url );
+		$home_url = untrailingslashit( $home_url );
+
+		return $home_url;
+	}
+
+	/**
+	 * To remove home url from the content.
+	 *
+	 * @since 2.2
+	 *
+	 * @param string $content Content from home_url need to remove.
+	 *
+	 * @return string Content after removing home_url.
+	 */
+	public static function remove_domain( $content ) {
+
+		if ( empty( $content ) ) {
+			return '';
+		}
+
+		$home_url = static::get_home_url();
+		$home_url = str_replace( [ '.', '/' ], [ '\.', '\\\\{1,5}\/' ], $home_url );
+
+		/**
+		 * Reference: https://regex101.com/r/c25pNF/1
+		 */
+		$regex = "/http[s]?:\\\\{0,5}\/\\\\{0,5}\/$home_url/mU";
+
+		$content = preg_replace( $regex, '', $content );
+
+		return $content;
+	}
+
+	/**
+	 * To generate hash of object.
+	 *
+	 * @since 2.2
+	 *
+	 * @param string|array|object $object Object for that hash need to generate.
+	 *
+	 * @return string Hash value of provided object.
+	 */
+	public static function generate_hash( $object ) {
+
+		if ( empty( $object ) ) {
+			return '';
+		}
+
+		if ( is_object( $object ) ) {
+			$object = (array) $object;
+		}
+
+		if ( is_array( $object ) ) {
+			ksort( $object );
+			$object = wp_json_encode( $object );
```

Comment:
I was sure there was a `JSON_*` constant to `json_encode()` to sort key. But I don't see it.